### ISSUE TITLE
[BUG] Custom CSS classes are being reported by the `no-custom-classname` rule.

### ISSUE THREAD

**Issue by robert-barahona** [francoismassart/eslint-plugin-tailwindcss#458]
**Describe the bug**
Custom CSS classes that are used or imported by `main.css` are being detected by the `no-custom-classname` rule. However, Tailwind 4 does allow to use custom CSS, so they should be ignored and not reported by the rule.

**To Reproduce**

1. Clone this [repo](https://github.com/robert-barahona/next-app)
2. Install deps `pnpm install`
3. Run `pnpm eslint`
4. See the error in the terminal

**Expected behavior**
`custom-gradient` should be ignored, and no ESLint error should be reported.

**Screenshots**
<img width="955" height="197" alt="Image" src="https://github.com/user-attachments/assets/0aa75dbb-bfe3-461e-bd25-878ce78fd492" />

**Environment (please complete the following information):**

- OS: macOs
- Softwares + version used:
  - eslint 9
  - tailwindcss 4
  - eslint-plugin-tailwindcss 4.0.3

**Additional context**

```js
// eslint.config.mjs
...
{
	plugins: {
		tailwindcss,
	},
	settings: {
		tailwindcss: {
			cssConfigPath: "./app/styles/main.css",
		},
	},
	rules: {
		"tailwindcss/no-custom-classname": "error",
	},
},
...
```

```css
/* main.css */
@import "tailwindcss";

.custom-gradient {
  background: linear-gradient(
    133.67deg,
    #d6409f -2.38%,
    #ab4aba 49.91%,
    #8e4ec6 100.06%
  );
}
```

**Eslint config file or live demo:**
[https://github.com/robert-barahona/next-app](https://github.com/robert-barahona/next-app)

**Comment by francoismassart** [comment]
In order to be "picked up" by the Tailwind CSS compiler (which is used by this rule), the declaration of your rule must be changed to

```css
@import "tailwindcss";

@utility custom-gradient {
  background: linear-gradient(
    133.67deg,
    #d6409f -2.38%,
    #ab4aba 49.91%,
    #8e4ec6 100.06%
  );
}
```

It works with `@utility`

<img width="552" height="164" alt="Image" src="https://github.com/user-attachments/assets/8981d496-fe55-454b-8f40-3350bfd6a2a0" />

and I've also tried to use the layers but w/o success.

<img width="200" height="50" alt="Image" src="https://github.com/user-attachments/assets/e25a4065-cd00-45dd-b338-503297328c29" />

Here is a Tailwind CSS Play https://play.tailwindcss.com/9ygOFQ6oGb 
Even if the `@layer` generates some CSS, it is not "related" for the compiler.

**Comment by robert-barahona** [comment]
@francoismassart Thanks for your reply.

I'm aware of this, but it would be great if the rule worked without having to add the `@utility` directive.

Because in our case, we have some custom CSS that shouldn't be considered "utility" styles, since they're only used for a specific element.

**Comment by francoismassart** [comment]
As I'm not planning to work on this specific issue I'm closing it.

I added others test in https://play.tailwindcss.com/4t1DQjy680 and the "pure native" classnames (outside of tailwind) are not handled by the official intellisense as well...
